### ISSUE TITLE
Invoice sync

### ISSUE THREAD

**Issue by schnere** [openlabs/magento_integration#107]
Hi,

I did not see a function to sync invoices. Is there any plan to integrate this?

Thanks

**Comment by sharoonthomas** [comment]
@schnere How would you envision this integration to work ? Can you explain the expectations from this sync ?

**Comment by schnere** [comment]
@sharoonthomas Thanks for your fast answer. 

I think, the module should check if an invoice is created anywhere. Or better trigger an event on invoice creation. Then the invoice data should be synced to the other system.
So if an invoice is created in Magento, the Magento module should trigger invoice creation with the same data (possible partial invoicing) in OpenERP. 

There are a few things to consider:
- Same invoice number in OpenERP and Magento:
  As with sales order number always the Magento invoice number should be used with a specified prefix.
  So for Magento orders the OpenERP "create invoice" button should trigger Magento invoice creation which then triggers the OpenERP invoice creation.
- Different invoice templates in Magento and OpenERP:
  Here I think the OpenERP invoice pdf should be transferred to Magento or if the invoice gets printed in Magento, the OpenERP invoice should be fetched and printed.

As OpenERP is the base financial system, all invoices should get there. But for orders placed in a Magento store, the customer would have the ability to see the invoice in his customer account. So there are also reasons to have invoices in Magento.

I might also be able to help on implementing this.

Best regards,

schnere

**Comment by capitaine** [comment]
I don't think this feature should be implemented this way. As said by schnere above, OpenERP is the ++main++ financial system. This means it can handle much more complex operations (financial or accounting) than Magento can do. Magento has a very simplified approach of accounting. For example, it creates systematically an invoice when registering an order with a valid payment. You may not wish to create an invoice in your accounting system at the ordering time but later when delivery is ready. Magento records every payment as a payment. In openErp (and in accounting science) a payment may be considered as a payment or as a deposit, it's not exactly the same. Also, you may differ the credit card payment some days till the order is prepared and ready for delivery. Or you may have to partially cancel an order or even create multiple invoices for a single order (if multiple deliveries when not all products of the order are available at the same time). So there are a lot of reasons for not creating automatically an invoice inside openErp when Magento creates one. From my point of view, synchronization of invoices should go the other way: from openErp to Magento. This would require an extension (Magento side) that would prevent magento to emit invoices up to OpenERP does it.

**Comment by schnere** [comment]
@capitaine You are right, the invoices should be generated in OpenERP and then synched back to Magento, so that the customer has the ability to see the invoice in his Magento customer account.

But tax sync should be fixed first, so that invoice would get correct tax settings.

**Comment by shalabhaggarwal** [comment]
The tax sync has been fixed for simple products and shipping (See #82 ).
